Infrastructure/Security/UserAccesor.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Application.Interfaces;
+using Microsoft.AspNetCore.Http;
+using System.Security.Claims;
+
+namespace Infrastructure.Security
+{
+    public class UserAccesor : IUserAccesor
+    {
+        //because we're not inside the context of our API, we need to access the http context via this interface
```

Comment:
Do you need this comment here? 
If this information is relevant to you, I advice you to write this class code in a way that it doesn't need any extra explanation from comments

Reply:
I removed the comment.